log/value.go

```diff
@@ -39,6 +39,36 @@ func Timestamp(t func() time.Time) Valuer {
 	return func() interface{} { return t() }
 }
 
+// TimestampFormat returns a Valuer that produces a TimeFormat value from
+// layout and the time returned by t. Users will probably want to use
+// DefaultTimestamp or DefaultTimestampUTC.
+func TimestampFormat(t func() time.Time, layout string) Valuer {
+	return func() interface{} {
+		return TimeFormat{
+			Time:   t(),
+			Layout: layout,
+		}
+	}
+}
+
+// A TimeFormat represents an instant in time and a layout used when
+// marshaling to a text format.
+type TimeFormat struct {
+	Time   time.Time
+	Layout string
+}
+
+func (tf TimeFormat) String() string {
+	return tf.Time.Format(tf.Layout)
+}
+
+// MarshalText implements encoding.TextMarshaller.
+func (tf TimeFormat) MarshalText() (text []byte, err error) {
+	b := make([]byte, 0, len(tf.Layout)+10)
```

Comment:
Oh! Well, if the magic number is OK there, I reckon it's OK here too. Nevermind my request.